### PR TITLE
hypre build with netlib-lapack

### PR DESCRIPTION
**Pull request by eklee15**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hypre/package.py [hunk 1]
```diff
@@ -69,6 +69,12 @@ def install(self, spec, prefix):
             '--with-blas-libs=%s' % ' '.join(blas.names),
             '--with-blas-lib-dirs=%s' % ' '.join(blas.directories)
         ]
+        
+        if spec.satisfies('^netlib-lapack'):
+            # netlib-lapack puts the libs in lib64
+            configure_args.extend([
+                "--with-lapack-lib-dirs=%s/lib64".join(lapack.directories),
```
Comment (davydden) [inline var/spack/repos/builtin/packages/hypre/package.py]: that's exactly what should happen above. It's the wrong place to fix. If you `netlib-lapack` lives in `lib64`, you need to fix that in `netlib-lapack` package.

**Comment by davydden** [comment]
I am closing this as I feel strongly that it's a wrong way to fix the issue.

File: var/spack/repos/builtin/packages/hypre/package.py (hunk 1 above)
Comment (eklee15) [inline var/spack/repos/builtin/packages/hypre/package.py]: Can you explain how to do this in netlib-lapack package? It seems I have to use setup_dependent_package right?

Comment (davydden) [inline var/spack/repos/builtin/packages/hypre/package.py]: No, you need look at how blas/lapack libs are being found inside netlib-lapack . Compare this to your directory stracture and adjust the code if needed

Comment (davydden) [inline var/spack/repos/builtin/packages/hypre/package.py]: @eklee15 look at https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/netlib-lapack/package.py#L73-L85

it at looks good to me! So you need to debug and see what value do you get for `lapack.directories` and `blas.directories` and why `hypre` does not like it. It all should be working OOB.

**Comment by davydden** [comment]
@eklee15 proof that `hypre` builds in general with `netlib-lapack`:
```
$ spack install hypre^netlib-lapack
==> Installing hypre
==> netlib-lapack is already installed in /Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/netlib-lapack-3.6.1-yipa2juilp2q4knrj4rnhrwggugizpbj
==> openmpi is already installed in /Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/openmpi-2.0.2-oos4cxprn2bislzhc3rbc3lo4dadginw
==> Using cached archive: /Users/davydden/spack/var/spack/cache/hypre/hypre-2.11.1.tar.gz
==> Staging archive: /Users/davydden/spack/var/spack/stage/hypre-2.11.1-akikjekh2bzm6ast7hersm35qd4h3efq/hypre-2.11.1.tar.gz
==> Created stage in /Users/davydden/spack/var/spack/stage/hypre-2.11.1-akikjekh2bzm6ast7hersm35qd4h3efq
==> No patches needed for hypre
==> Building hypre [Package]
==> Executing phase : 'install'
==> Successfully installed hypre
  Fetch: 0.03s.  Build: 1m 20.79s.  Total: 1m 20.82s.
[+] /Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/hypre-2.11.1-akikjekh2bzm6ast7hersm35qd4h3efq
```

**Comment by davydden** [comment]
here's blas/lapack parts of the configure flags which were correctly used for me:
```
'--with-lapack-libs=lapack' 
'--with-lapack-lib-dirs=/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/netlib-lapack-3.6.1-yipa2juilp2q4knrj4rnhrwggugizpbj/lib' 
'--with-blas-libs=blas' 
'--with-blas-lib-dirs=/Users/davydden/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/netlib-lapack-3.6.1-yipa2juilp2q4knrj4rnhrwggugizpbj/lib'
```
in your case, you should see `*-lib-dirs` ending with `lib64` if `netlib-lapack` installed the libs into `prefix/lib64`.

**Pull request closed without merging**